V 01

vladsadretdinov opened the pull request.

vladsadretdinov committed:
V-01 Connect fonts

vladsadretdinov committed:
Connect normalize.css

vladsadretdinov committed:
V-01, F-01: Finish homepage layout

FFusion commented on src/app/app.component.html:
```diff
@@ -1 +1,295 @@
+
+<div class="app">
+  <aside class="aside">
+    <svg
```
все svg нужно вынести в отдельные файлы

FFusion commented on src/app/app.component.html:
```diff
@@ -1 +1,295 @@
-<h1>Проверка стилей</h1>
+<!-- имя-блока__имя-элемента_имя-модификатора_значение-модификатора. -->
```
комментарии нужно удалять

FFusion commented on src/app/app.component.html:
```diff
@@ -1 +1,295 @@
+    <span class="aside__language" tabindex="0"> Eng </span>
+  </aside>
+
+  <nav [ngClass]="{ nav: true, nav_closed: this.menu_show }">
```
[ngClass]="{ nav: true } - в этом случае лучше указать class и не тащить nav в ngClass

FFusion commented on src/app/app.component.html:
```diff
@@ -1 +1,295 @@
+  <nav [ngClass]="{ nav: true, nav_closed: this.menu_show }">
+    <aside class="aside">
+      <img
+        (click)="this.menu_click()"
```
this в шаблоне не нужен. Фунциии нужно называть в формате camelCase

FFusion commented on src/app/app.component.ts:
```diff
@@ -1,9 +1,37 @@
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss'],
+  // Temporary solution in favor of BEM
+  encapsulation: ViewEncapsulation.None,
```
очень плохая идея так делать, чстили желательно изолировать в рамках компонента

FFusion commented on src/app/app.component.html:
```diff
@@ -1 +1,295 @@
-<h1>Проверка стилей</h1>
```
нужно разбить на отдельные компоненты. Делать все в app.component плохая идея. Получился слишком большой файл со стилями и много логики в шаблоне.

vladsadretdinov committed:
V-01: Remove META datum

vladsadretdinov committed:
V-01: 1. Svg to own file
2. Delete comments
3. Delete 'this' in template
4. camelCase func naming
5. Dont use ngClass if class is always expected

vladsadretdinov committed:
V-01: Make own button component

vladsadretdinov committed:
V-01: Break monoblock into components

FFusion commented on src/app/components/button/button.component.ts:
```diff
@@ -0,0 +1,24 @@
+  templateUrl: './button.component.html',
+  styleUrls: ['./button.component.scss'],
+})
+export class ButtonComponent implements OnInit {
```
button нужно вынести либо в отдельный модуль, либо в shared модуль. Наверняка он будет использоваться в остальных задачах. Это необязательно делать сейчас

FFusion commented on src/app/components/button/button.component.ts:
```diff
@@ -0,0 +1,24 @@
+export class ButtonComponent implements OnInit {
+  @Input() size: 'large' | 'medium' | undefined = undefined;
+
+  @Input() theme: 'green' | 'turquoise' | 'red' | 'purple' | undefined =
```
лучше по умолчанию задать size и theme, undefined вообще не использовать

FFusion commented on src/app/components/button/button.component.ts:
```diff
@@ -0,0 +1,24 @@
+  classList: string = '';
+
+  ngOnInit(): void {
+    if (this.size) {
```
если мы убираем undefined - условие не нужно

FFusion commented on src/app/app.component.html:
```diff
@@ -1,841 +1,83 @@
+    <span class="aside__language" tabindex="0"> Eng </span>
+  </aside>
+
+  <nav class="nav" [ngClass]="{ nav_closed: menu_show }">
```
давай nav и aside тоже вынесем в отдельный компонент, потому что app.component еще достаточно перегружен, стилей много

FFusion commented on src/app/app.component.ts:
```diff
@@ -6,5 +6,9 @@ import { Component } from '@angular/core';
+  menu_show = true;
+
+  menuClick() {
+    this.menu_show = !this.menu_show;
```
все переменные и названия функций - camelCase

FFusion commented on src/app/components/slider/slider.component.html:
```diff
@@ -0,0 +1,106 @@
+<input
```
давай попробуем вынести инфу в константы и сделать через ngFor

FFusion commented on src/app/components/slider/slider.component.html:
```diff
@@ -0,0 +1,106 @@
+  </div>
+
+  <ul class="slider__content">
+    <li class="slider__item" id="slide1">
```
слайды тоже через ngFor

FFusion commented on src/app/components/slider/slider.component.html:
```diff
@@ -0,0 +1,106 @@
+    <li class="slider__item" id="slide4">
+      <h2 class="slider__title">Обслуживание</h2>
+      <p class="slider__description">Автомобиль проходит еженедельное ТО.</p>
+      <app-button [size]="'medium'" [theme]="'purple'"> Подробнее </app-button>
```
можно проще, чтобы было меньше кавычек `<app-button size="medium" theme="purple"> Подробнее </app-button>`

FFusion commented on src/app/components/slider/slider.component.ts:
```diff
@@ -0,0 +1,29 @@
+export class SliderComponent implements OnDestroy {
+  slide = '1';
+
+  intervalId = setInterval(this.sliderClick.bind(this), 5000, 'next', true);
```
const slideInterval = 5000; константы нужно вынести в отдельный файл slider.const.ts и импортировать в компонент

FFusion commented on src/app/components/slider/slider.component.ts:
```diff
@@ -0,0 +1,29 @@
+
+  sliderClick(dir = 'next', auto = false) {
+    if (auto !== true) {
+      clearInterval(this.intervalId);
```
if (!auto) {} - так короче

FFusion commented on src/app/components/slider/slider.component.ts:
```diff
@@ -0,0 +1,29 @@
+    }
+    if (dir === 'next') {
+      const newValue = +this.slide + 1;
+      this.slide = String(newValue > 4 ? 1 : newValue);
```
номера слайдов лучше сделать через enum, чтобы очистить компонент от чисел

FFusion commented on src/app/components/button/button.component.ts:
```diff
@@ -0,0 +1,24 @@
+@Component({
+  selector: 'app-button',
+  templateUrl: './button.component.html',
+  styleUrls: ['./button.component.scss'],
```
в компоненты желательно можно добавлять onPush стратегию, для начала хотя бы в простые

vladsadretdinov committed:
V-01: 1. Make shared module
2. Use default type for button inputs
3. Variable names to camelCase style
4. Make slider own const vars
5. Set onPush strategy to dumb components

vladsadretdinov committed:
V-01: Remake slider to more dynamic version

vladsadretdinov committed:
V-01: Make more components

FFusion commented on src/app/components/aside/aside.component.html:
```diff
@@ -0,0 +1,24 @@
+<aside class="aside">
+  <svg
```
давай подумаем как оптимизировать можно, фактически у нас меняется только use, может svg оставим одну?

FFusion commented on src/app/components/slider/slider.component.ts:
```diff
@@ -21,12 +23,12 @@ export class SliderComponent implements OnDestroy {
       clearInterval(this.intervalId);
     }
     if (dir === 'next') {
```
давай вместо 'next' и 'prev' заведем boolean переменную

vladsadretdinov committed:
V-01: Close all problems
